### ISSUE TITLE
Some suggestions for `ggpredict()`

### ISSUE THREAD

**Issue by tqtbui** [strengejacke/ggeffects#320]
I just want to have some suggestions for the function `ggpredict()` and the related. 

1. In the note for `vcov.fun` for example, it says "Note that probably not all model objects that work with `ggpredict()` are also supported by the `sandwich` or `clubSandwich-package`". If that is the case then why don't we have an option for users to supply their own `vcov`?, for example, I can have a vcov derived from a new method. 

2. The way to specify `terms` right now is not function-friendly. If I want to specify specific values for a variable, I will need to specify the name of a specific vector that contains the variable.  From the documentation, e.g., define a vector `v = c(1000, 2000, 3000)` and then use `terms = "income [v]"`. However, now `v` has to be a global variable, and if it is supplied by an user within another function that calls `ggpredict()`, it does not work. 

Thank you, I hope my suggestions are considered for future updates.

**Comment by strengejacke** [comment]
Do you have a reproducible example for 2)?

Not sure if this has been fixed in the recent CRAN update:

``` r
library(ggeffects)
data(efc)
foo <- function(data) {
  fit <- lm(barthtot ~ c12hour + c172code, data = data)
  v <- c(20, 50, 70)
  ggpredict(fit, terms = "c12hour [v]")
}
foo(efc)
#> # Predicted values of Total score BARTHEL INDEX
#> 
#> c12hour | Predicted |         95% CI
#> ------------------------------------
#>      20 |     71.31 | [69.41, 73.22]
#>      50 |     62.36 | [60.60, 64.12]
#>      70 |     56.39 | [54.40, 58.37]
#> 
#> Adjusted for:
#> * c172code = 1.97
```

<sup>Created on 2023-05-12 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by strengejacke** [comment]
Thanks, should be fixed in #326 

``` r
library(ggeffects)
data(efc)

fn1 <- function(data) {
  fit <- lm(barthtot ~ c12hour + neg_c_7 + c161sex + c172code, data = data)
  v <- c(20, 50, 70)
  ggpredict(fit, terms = "c12hour [20, 50, 70]")
}

fn2 <- function(data) {
  fit <- lm(barthtot ~ c12hour + neg_c_7 + c161sex + c172code, data = data)
  v <- c(20, 50, 70)
  ggpredict(fit, terms = "c12hour [v]")
}

fn3 <- function(data, v) {
  fit <- lm(barthtot ~ c12hour + c172code, data = data)
  ggpredict(fit, terms = "c12hour [v]")
}

fn1(efc)
#> # Predicted values of Total score BARTHEL INDEX
#> 
#> c12hour | Predicted |         95% CI
#> ------------------------------------
#>      20 |     70.38 | [68.56, 72.19]
#>      50 |     62.78 | [61.11, 64.45]
#>      70 |     57.71 | [55.81, 59.62]
#> 
#> Adjusted for:
#> *  neg_c_7 = 11.84
#> *  c161sex =  1.76
#> * c172code =  1.97

fn2(efc)
#> # Predicted values of Total score BARTHEL INDEX
#> 
#> c12hour | Predicted |         95% CI
#> ------------------------------------
#>      20 |     70.38 | [68.56, 72.19]
#>      50 |     62.78 | [61.11, 64.45]
#>      70 |     57.71 | [55.81, 59.62]
#> 
#> Adjusted for:
#> *  neg_c_7 = 11.84
#> *  c161sex =  1.76
#> * c172code =  1.97

v <- c(5, 6, 7)
a <- c(0.3, 0.5, 0.7)
fn3(efc, a)
#> # Predicted values of Total score BARTHEL INDEX
#> 
#> c12hour | Predicted |         95% CI
#> ------------------------------------
#>    0.30 |     77.20 | [74.93, 79.46]
#>    0.50 |     77.14 | [74.87, 79.40]
#>    0.70 |     77.08 | [74.82, 79.33]
#> 
#> Adjusted for:
#> * c172code = 1.97


fn1(efc)
#> # Predicted values of Total score BARTHEL INDEX
#> 
#> c12hour | Predicted |         95% CI
#> ------------------------------------
#>      20 |     70.38 | [68.56, 72.19]
#>      50 |     62.78 | [61.11, 64.45]
#>      70 |     57.71 | [55.81, 59.62]
#> 
#> Adjusted for:
#> *  neg_c_7 = 11.84
#> *  c161sex =  1.76
#> * c172code =  1.97
fn2(efc)
#> # Predicted values of Total score BARTHEL INDEX
#> 
#> c12hour | Predicted |         95% CI
#> ------------------------------------
#>      20 |     70.38 | [68.56, 72.19]
#>      50 |     62.78 | [61.11, 64.45]
#>      70 |     57.71 | [55.81, 59.62]
#> 
#> Adjusted for:
#> *  neg_c_7 = 11.84
#> *  c161sex =  1.76
#> * c172code =  1.97
fn3(efc, a)
#> # Predicted values of Total score BARTHEL INDEX
#> 
#> c12hour | Predicted |         95% CI
#> ------------------------------------
#>    0.30 |     77.20 | [74.93, 79.46]
#>    0.50 |     77.14 | [74.87, 79.40]
#>    0.70 |     77.08 | [74.82, 79.33]
#> 
#> Adjusted for:
#> * c172code = 1.97
```

<sup>Created on 2023-05-12 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>
